app/src/main/java/us/shandian/giga/service/DownloadManagerService.java

```diff
@@ -353,28 +355,29 @@ public void updateForegroundState(boolean state) {
      * @param kind         type of file (a: audio  v: video  s: subtitle ?: file-extension defined)
      * @param threads      the number of threads maximal used to download chunks of the file.
      * @param psName       the name of the required post-processing algorithm, or {@code null} to ignore.
-     * @param source       source url of the resource
+     * @param streamInfo   stream metadata that may be written into the downloaded file.
      * @param psArgs       the arguments for the post-processing algorithm.
      * @param nearLength   the approximated final length of the file
      * @param recoveryInfo array of MissionRecoveryInfo, in case is required recover the download
      */
     public static void startMission(Context context, String[] urls, StoredFileHelper storage,
-                                    char kind, int threads, String source, String psName,
+                                    char kind, int threads, StreamInfo streamInfo, String psName,
                                     String[] psArgs, long nearLength,
                                     ArrayList<MissionRecoveryInfo> recoveryInfo) {
         final Intent intent = new Intent(context, DownloadManagerService.class)
                 .setAction(Intent.ACTION_RUN)
                 .putExtra(EXTRA_URLS, urls)
                 .putExtra(EXTRA_KIND, kind)
                 .putExtra(EXTRA_THREADS, threads)
-                .putExtra(EXTRA_SOURCE, source)
+                .putExtra(EXTRA_SOURCE, streamInfo.getUrl())
```

Comment:
Is the source field still needed? It feels like it is always derieved from the streaInfo.